src/Functions/stringToMap.cpp

```diff
@@ -0,0 +1,242 @@
+    extern const int NUMBER_OF_ARGUMENTS_DOESNT_MATCH;
+    extern const int BAD_ARGUMENTS;
+}
+
+class FunctionMapFromString : public IFunction
+{
+public:
+    static constexpr auto name = "mapFromString";
+    static FunctionPtr create(ContextPtr) { return std::make_shared<FunctionMapFromString>(); }
+
+    String getName() const override { return name; }
+
+    bool isVariadic() const override { return true; }
+    size_t getNumberOfArguments() const override { return 0; }
+
+    bool isSuitableForShortCircuitArgumentsExecution(const DataTypesWithConstInfo & /*arguments*/) const override { return true; }
+
+    DataTypePtr getReturnTypeImpl(const ColumnsWithTypeAndName & arguments) const override
+    {
+        FunctionArgumentDescriptors args{
+            {"str_expr", &isStringOrFixedString<IDataType>, nullptr, "String or Fixed String"}
+        };
+        FunctionArgumentDescriptors optional_args{
+            {"pair_delim", &isString<IDataType>, isColumnConst, "const String"},
+            {"key_value_delim", &isString<IDataType>, isColumnConst, "const String"}
+        };
+
+        validateFunctionArgumentTypes(*this, arguments, args, optional_args);
+
+        DataTypePtr string_type = std::make_shared<DataTypeString>();
+        DataTypePtr string_type_nullable = std::make_shared<DataTypeNullable>(string_type);
+        DataTypes key_value_type{string_type, string_type_nullable};
+
+        return std::make_shared<DataTypeMap>(key_value_type);
+    }
+
+    bool useDefaultImplementationForConstants() const override { return true; }
+
+    static void executeString(const ColumnString * col_str_in, ColumnPtr & col_res, size_t input_rows_count, char pair_delim, char key_value_delim)
+    {
+        String pair_delim_str(1, pair_delim);
+        
+        const ColumnString::Chars & in_vec = col_str_in->getChars();
+        const ColumnString::Offsets & in_offsets = col_str_in->getOffsets();
+        size_t size = in_offsets.size();
+
+        MutableColumnPtr keys_data = DataTypeString().createColumn();
+        MutableColumnPtr values_data = DataTypeString().createColumn();
+        MutableColumnPtr offsets = DataTypeNumber<IColumn::Offset>().createColumn();
+
+        keys_data->reserve(input_rows_count);
+        values_data->reserve(input_rows_count);
+        offsets->reserve(input_rows_count);
+
+        ColumnUInt8::MutablePtr values_null_map = ColumnUInt8::create();
+
+        ColumnString::Offset curr_res_offset = 0, curr_in_offset = 0;
+
+        std::vector<String> key_value_pairs;
+
+        for (size_t i = 0; i < size; ++i)
+        {
+            key_value_pairs.clear();
+            const char * pos_in = reinterpret_cast<const char *>(&in_vec[curr_in_offset]);
+            size_t curr_size = in_offsets[i] - curr_in_offset - 1;
+            // splitting the string into key-value pairs
+            boost::split(key_value_pairs, std::string(pos_in, curr_size), boost::is_any_of(pair_delim_str));
```

Comment:
Could you please check if we can use `std::string_view` instead of `std::string` in l. 93? This would save another heap allocation.